inc/msp/msp_msg.hpp

```diff
@@ -82,933 +114,5172 @@ static const std::vector<std::string> FEATURES = {
     "BLACKBOX", "CHANNEL_FORWARDING", "TRANSPONDER", "OSD"
 };
 
+
+enum class PID_Element: uint8_t {
+    PID_ROLL,
+    PID_PITCH,
+    PID_YAW,
+    PID_POS_Z,
+    PID_POS_XY,
+    PID_VEL_XY,
+    PID_SURFACE,
+    PID_LEVEL,
+    PID_HEADING,
+    PID_VEL_Z,
+    PID_ITEM_COUNT
+};
+
 /////////////////////////////////////////////////////////////////////
 /// Cleanflight
 
 // MSP_API_VERSION: 1
-struct ApiVersion : public Request {
-    ID id() const { return ID::MSP_API_VERSION; }
-
-	size_t protocol;
-	size_t major;
-	size_t minor;
-
-    void decode(const std::vector<uint8_t> &data) {
-        protocol = data[0];
-        major = data[1];
-        minor = data[2];
+struct ApiVersion : public Message {
+    ApiVersion(FirmwareVariant v) : Message(v) {}
+    
+    virtual ID id() const override { return ID::MSP_API_VERSION; }
```

Comment:
`id` and others are virtual because the class inherits from the virtual method defined in the parent class.

Reply:
That `Message::id()` is `virtual` makes sense since you want inherited classes (e.g. `ApiVersion`) to provide their own implementation of `id()`. But does the `ApiVersion::id()` also need to be `virtual`? I.e. do you expect that there will be a message definition that is derived from `ApiVersion` and that needs to reimplement `id()`?